src/CommunityToolkit.Maui.Core/Interfaces/IDockLayout.shared.cs

```diff
@@ -0,0 +1,31 @@
+﻿namespace CommunityToolkit.Maui.Core.Interfaces;
+
+/// <summary>Docking position for a view in <see cref="IDockLayout"/>.</summary>
+public enum DockEnum
+{
+	/// <summary>Do not dock.</summary>
+	None = 0,
+	/// <summary>Dock at the top.</summary>
+	Top = 1,
+	/// <summary>Dock to the left.</summary>
+	Left = 2,
+	/// <summary>Dock to the right.</summary>
+	Right = 3,
+	/// <summary>Dock at the bottom.</summary>
+	Bottom = 4
+}
+
+/// <summary>DockLayout positions its child elements along the edges of the layout container.</summary>
+public interface IDockLayout : ILayout
+{
+	/// <summary>If true, the last child fills the remaining space.</summary>
+	public bool LastChildFill { get; }
+
+	/// <summary>Horizontal (width) and vertical (height) spacing between docked views.</summary>
+	public SizeF Spacing { get; }
```

Comment:
Let's use `double` instead of `SizeF` for `Spacing`.

Using `double` aligns better with the spacing properties inside of other Layouts:
- `public double Grid.ColumnSpacing { get; set; }`,
- `public double Grid.RowSpacing { get; set; }`
- `public double VerticalStackLayout.Spacing { get; set; }`
- `public double HorizontalStackLayout.Spacing { get; set; }`

This'll be used as vertical spacing when `DockPosition.Left` and `DockPosition.Right`, and it'll be used for horizontal spacing when `DockPosition.Top` and `DockPosition.Bottom`, similar to how `StackLayout` auto-implements the Spacing based upon `StackLayout.Orientation`.

```suggestion
	public double Spacing { get; }
```

Reply:
> This'll be used as vertical spacing when DockPosition.Left and DockPosition.Right, and it'll be used for horizontal spacing when DockPosition.Top and DockPosition.Bottom.

The way it's implemented now, you can have different spacing for horizontal/vertical and for my purposes I use that a lot. I would therefore suggest having both `double VerticalSpacing` and `double HorizontalSpacing`.